### PR TITLE
[seahub] Added the ability to modify the format of the text preview s…

### PR DESCRIPTION
**Pull request by zMingGit**
…uffix

### CODE REVIEW

**Comment by shoeper** [comment]
Looks like this requires to have the list. So everyone upgrading needs to add it to his config. Would be better to keep the old list as fallback in case nothing was defined in seahub_settings.py

File: seahub/settings.py [hunk 1]
```diff
@@ -685,4 +685,6 @@ def load_local_settings(module):
     'SHARE_LINK_TOKEN_LENGTH': (SHARE_LINK_TOKEN_LENGTH, ''),
     'SHARE_LINK_PASSWORD_MIN_LENGTH': (SHARE_LINK_PASSWORD_MIN_LENGTH,''),
     'ENABLE_TWO_FACTOR_AUTH': (ENABLE_TWO_FACTOR_AUTH,''),
+
+    'TEXT_PREVIEW_EXT': ('ac, am, bat, c, cc, cmake, cpp, cs, css, diff, el, h, html, htm, java, js, json, less, make, org, php, pl, properties, py, rb, scala, script, sh, sql, txt, text, tex, vi, vim, xhtml, xml, log, csv, groovy, rst, patch, go', ''),
```
Comment (xiez) [inline seahub/settings.py]: 新增变量 TEXT_PREVIEW_EXT , 方便别人在 seahub_settings.py 里覆盖

File: seahub/utils/__init__.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def is_cluster_mode():
 MAX_INT = 2147483647
 
 PREVIEW_FILEEXT = {
-    TEXT: ('ac', 'am', 'bat', 'c', 'cc', 'cmake', 'cpp', 'cs', 'css', 'diff', 'el', 'h', 'html', 'htm', 'java', 'js', 'json', 'less', 'make', 'org', 'php', 'pl', 'properties', 'py', 'rb', 'scala', 'script', 'sh', 'sql', 'txt', 'text', 'tex', 'vi', 'vim', 'xhtml', 'xml', 'log', 'csv', 'groovy', 'rst', 'patch', 'go'),
+    #TEXT: ('ac', 'am', 'bat', 'c', 'cc', 'cmake', 'cpp', 'cs', 'css', 'diff', 'el', 'h', 'html', 'htm', 'java', 'js', 'json', 'less', 'make', 'org', 'php', 'pl', 'properties', 'py', 'rb', 'scala', 'script', 'sh', 'sql', 'txt', 'text', 'tex', 'vi', 'vim', 'xhtml', 'xml', 'log', 'csv', 'groovy', 'rst', 'patch', 'go'),
```
Comment (xiez) [inline seahub/utils/__init__.py]: 去掉注释

File: seahub/utils/__init__.py [hunk 3]
```diff
@@ -334,13 +334,27 @@ def get_user_repos(username, org_id=None):
...
+    return []
+
 def get_file_type_and_ext(filename):
     """
     Return file type and extension if the file can be previewd online,
     otherwise, return unknown type.
     """
     fileExt = os.path.splitext(filename)[1][1:].lower()
     filetype = FILEEXT_TYPE_MAP.get(fileExt)
+    if str(fileExt) in get_conf_text_ext():
```
Comment (xiez) [inline seahub/utils/__init__.py]: 去掉 str，这行移到上面行

File: seahub/views/file.py [hunk 4]
```diff
@@ -328,7 +329,7 @@ def can_preview_file(file_name, file_size, repo=None):
     if repo and repo.encrypted and (file_type in (DOCUMENT, SPREADSHEET, PDF)):
         return (False, _(u'The library is encrypted, can not open file online.'))
 
-    if file_ext in FILEEXT_TYPE_MAP:  # check file extension
+    if file_ext in FILEEXT_TYPE_MAP or file_ext in get_conf_text_ext():  # check file extension
```
Comment (xiez) [inline seahub/views/file.py]: 新写一个函数来判断

File: seahub/settings.py [hunk 5]
```diff
@@ -685,4 +685,7 @@ def load_local_settings(module):
...
 }
+TEXT_PREVIEW_EXT = ''
```
Comment (xiez) [inline seahub/settings.py]: 应该放到上面去，类似其他变量

File: seahub/utils/__init__.py [hunk 6]
```diff
@@ -334,12 +333,26 @@ def get_user_repos(username, org_id=None):
 
     return (owned_repos, shared_repos, groups_repos, public_repos)
 
+def get_conf_text_ext():
+    """
+    get the conf of text ext in constance settings, and remove space.
```
Comment (xiez) [inline seahub/utils/__init__.py]: Get

File: seahub/utils/__init__.py [hunk 7]
```diff
@@ -334,12 +333,26 @@ def get_user_repos(username, org_id=None):
...
+    """
+    if hasattr(config, 'TEXT_PREVIEW_EXT'):
+        test_ext = getattr(config, 'TEXT_PREVIEW_EXT').split(',')
+        for i in range(len(test_ext)):
```
Comment (xiez) [inline seahub/utils/__init__.py]: return [x.strip() for x in text_ext]

**Pull request merged**